### ISSUE TITLE
God Eater 2 1.3 - Mission Description Broken with PPSSPP 1.2

### ISSUE THREAD

**Issue by Toohpick** [hrydgard/ppsspp#8602]
I know this game is buggy, but this update breaks something simple that worked fine in older versions.
As GE2 1.4 was broken since #8043 got merged, I've been playing on 1.3. I updated PPSSPP to 1.2 and all random crashes present in older PPSSPP versions were gone, but the quest descriptions got broken.

Old PPSSPP versions
![npjh50832_00016](https://cloud.githubusercontent.com/assets/17336867/13176821/d6f79e22-d70c-11e5-8a83-768ed6dfc15e.jpg)

1.2
![npjh50832_00001](https://cloud.githubusercontent.com/assets/17336867/13176944/cf8f8aa4-d70d-11e5-97d7-91f0f9e3f629.jpg)

Resetting settings to default didn't help. Note that this is with fast memory off since it breaks this game.

**Comment by daniel229** [comment]
Sound like the same cause by https://github.com/hrydgard/ppsspp/pull/8043 ,if I remove https://github.com/hrydgard/ppsspp/pull/8043 then using the hack make it work,the texts work fine.

**Comment by daniel229** [comment]
Even not revert https://github.com/hrydgard/ppsspp/pull/8043 ,then apply the hack the texts work fine.

**Comment by Toohpick** [comment]
Thank! Good to know it's possible to "fix" it without breaking other things. What hack should I apply? I couldn't figure out what triggers this issue exactly.

**Comment by unknownbrackets** [comment]
What's the hack that makes it work?

I guess it's having trouble allocating memory for the blue characters?

For that pull, I spent a lot of time dumping and comparing RAM to get the allocation more accurate.  We've still got inaccuracies, though, so probably we just need to fix more.

-[Unknown]

**Comment by daniel229** [comment]
```
bool fromTop = (nt.attr & PSP_THREAD_ATTR_LOW_STACK) == 0;
```

to
```
bool fromTop = 1;
```

**Comment by Toohpick** [comment]
@unknownbrackets Yeah, it fixed a lot of memory-related crashes, especially with savestates in some other games. The only inaccuracies I know are this issue and the known GE2 1.4 crash even without NPCs. 
@daniel229 I'll try that later, thanks. Just making sure, that's sceKernelThread.cpp right?

**Comment by daniel229** [comment]
@Toohpick Yes, in sceKernelThread.cpp,But have to revert https://github.com/hrydgard/ppsspp/pull/8043 first,otherwise it hangs.

**Comment by unknownbrackets** [comment]
Hmm, it'd be interesting to see the log (especially the most recent allocation / failed allocation) for both working (with hack + reverted 8043) and master (no hack + with 8043).

-[Unknown]

**Comment by daniel229** [comment]
Here is the logs
rename jpg to 7z

no hack + with 8043
![ppsspplog-clean](https://cloud.githubusercontent.com/assets/3481559/14415830/d4e52842-ffd5-11e5-8964-9eb2f32aeda3.jpg)

with hack + reverted 8043
![ppsspplog-hack](https://cloud.githubusercontent.com/assets/3481559/14415831/d4e74730-ffd5-11e5-85c1-51ad697e8bff.jpg)

**Comment by benderscruffy** [comment]
is this still an issue ??

**Comment by Deka710** [comment]
its also break the result screen, always shown 000000,tried the base game and 1.40 update didnt have this mission desc and result screen problem,only in 1.30 update:
![Screenshot_2023-05-28-08-09-26-338_org ppsspp ppssppgold](https://github.com/hrydgard/ppsspp/assets/97207704/84e9887e-4dfa-4aae-9e79-536ece1dcf3d)

**Comment by IrfanH495** [comment]
I tried it on software rendering.
several letters appear in random order, when selecting a mission.

https://github.com/hrydgard/ppsspp/assets/78854859/86e38c44-e6a0-44d1-9c66-7675ce6cbb7f

**Comment by Lumino5501** [comment]
> I tried it on software rendering some text is still missing. several letters appear in random order, when selecting a mission.
> 
>  mobizen_20240402_003735.mp4 
> my friend said the english patch version didn't experience it. I don't have the game to try. ![Screenshot_20240402-013027_PPSSPP](https://private-user-images.githubusercontent.com/78854859/318533041-18fe8256-9677-46fe-86ae-9e09059e1bf3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMzczMjAsIm5iZiI6MTcxMjAzNzAyMCwicGF0aCI6Ii83ODg1NDg1OS8zMTg1MzMwNDEtMThmZTgyNTYtOTY3Ny00NmZlLTg2YWUtOWUwOTA1OWUxYmYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAyVDA1NTAyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2NjNjMzAxNWE4OGQ5MjVkYzA2ODJlZDA4ZTI5YTMyN2MzOTcyZDI4MzZjM2FjNGEwODIwMGViYjM1NDg4OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.balzrnHBOBSJNfWoOqZqs9z-b5QIdD8mXk7zKjybR-E)

The random texts appearing is normal, it also happens in english patch, although we mostly dont notice it since it's not lag when not using software render.

What went missing though after the spacing issue was fixed is the colored aragami name texts
